src/Database/Type/DateTimeType.php

```diff
@@ -165,7 +169,7 @@ public function marshal($value)
                 $date = new $class($value);
                 $compare = true;
             }
-            if ($compare && $date && $date->format($this->_format) !== $value) {
+            if ($compare && $date && !$this->_compare($date, $value)) {
```

Comment:
Do you have an alternative PR for your recommendation? I am open to any other approach that would fix the issue.